좋아요 순으로 Content 정렬(내림차순)

hanurii opened the pull request:
- Paging 을 사용하여 간단하게 좋아요 순으로 목록을 출력할 수 있도록 만들었습니다

- 만들다가 든 생각은 트렌딩은 지금 현 시점의 트렌드한 게시물(좋아요 수가 많은 게시물)을 보여줘야 할텐데요.
- 지금 로직으로는 가장 오래된 포스트여도 좋아요 수가 많다면 트렌딩에 뜨게끔 해둔 것입니다.
- 기간같은 걸로 범위를 지정해준다음에 그 목록 속에서 좋아요 순으로 정렬하는 게 어떨까 하고 개인적으로 생각했습니다.
- 같이 이야기해보면 좋을 것 같습니다

hanurii committed:
[#8] Feat : 좋아요 순으로 Content 정렬 구현

- 포스트의 좋아요 수를 기준으로 내림차순 정렬하여 출력합니다.

hanurii committed:
[#8] Feat : 테스트 코드 작성 및 Spring Rest Docs 작성

- 필수적인 것들만 문서화를 진행했다
- optional 인 부분들을 명시해주는 것은 차후 추가할 예정

hanurii committed:
[#8] Modify : 메서드명 변경

- 한님과 서로 합의한 메서드 명으로 변경합니다

hanurii committed:
Merge branch 'dev' into feature/issue-008

hanurii committed:
[#8] Modify : 컴파일 오류가 생긴 부분 수정

hanurii committed:
[#8] Modify : 테스트 코드 수정

- 코드 병합 시 postService가 없어졌기 때문에 postController로 바꿔서 mokito 테스트 진행

hanurii committed:
[#8] Modify : 테스트 코드 메서드명 변경에 따른 adoc 문서 내용 변경

102092 commented on src/docs/asciidoc/api-guide.adoc:
```diff
@@ -79,3 +79,21 @@ include::{snippets}/post-controller-test/delete_post/http-response.adoc[]
 
 include::{snippets}/post-controller-test/delete_post/response-fields.adoc[]
+
+=== 좋아요 수를 기준으로 내림차순 출력 API
```
제가 이름을 위에서 `메인 페이지 조회(최신 순으로 정렬)` 이라 했는데,
혹시 이거랑 비슷하게 `메인 페이지 조회(좋아요 순으로 정렬)` 같이 해보는게 어떨까요?
아니면 혹시 좀 더 좋은 이름이 있으면 같이 이야기 해봐요!

hanurii replied:
통일된 이름이 좋을 것 같습니다 한님 의견을 수용할게요!

hanurii replied:
메인 페이지 조회(생성시간 순으로 내림차순 정렬)
좋아요 순으로 내림차순 정렬

102092 commented on src/main/java/com/codesquad/rare/domain/post/PostController.java:
```diff
@@ -23,6 +24,8 @@
 public class PostController {
 
   private final PostService postService;
+  private static final String DEFAULT_PAGE = "0";
+  private static final String DEFAULT_SIZE = "20";
```
현재 페이지 사이즈 수를, 저희가 정해주고 있는데 혹시 동적으로 그러니까 프론트에서 설정하는 것에 따라서 다르게 
페이지를 줄 수도 있을까요?
won이 그렇게 해주시길 원하셔서...

hanurii replied:
디폴트 페이지를 정해주는거고 프론트분들이 보내주시는데 page, size 값에 따라서 클라이언트 단에서 페이지 수를 조절할 수 있습니다!

102092 commented on src/main/java/com/codesquad/rare/domain/post/PostController.java:
```diff
@@ -44,4 +47,12 @@ public ApiResult delete(@PathVariable("id") Long postId) {
+  // 포스트를 좋아요 수를 기준으로 내림차순 출력
+  @GetMapping("/likes")
+  public ApiResult<List<Post>> findAllByLikesInDescendingOrder(
+      @RequestParam(value = "page", required = false, defaultValue = DEFAULT_PAGE) Integer page,
+      @RequestParam(value = "size", required = false, defaultValue = DEFAULT_SIZE) Integer size) {
```
`int` 가 아닌 wrapper class로 받는 이유가 있을까요?

hanurii replied:
nullable 하게 해주면 좋겠다는 생각을 먼저 했습니다. 혹시 @RequestParam으로 파라미터를 받을 때 어떻게 하시는 편이신가요?? 저도 많은 생각을 하고 타입을 정해주진 않았숩니다

102092 replied:
저는 primitive한 타입으로 그냥 받는 편입니당.. 혹시 null이 들어올 수 있는 곳에는 Optional 하게 받아서 처리하는게 좋지 않을까 생각이 들긴 하네요.

102092 commented on src/test/java/com/codesquad/rare/domain/post/PostControllerTest.java:
```diff
@@ -279,4 +284,95 @@ void find_post_by_id() throws Exception {
+    ResultActions result = this.mockMvc.perform(get("/posts/likes")
+        .contentType(MediaType.APPLICATION_JSON)
+        .accept(MediaType.APPLICATION_JSON)
+        .queryParams(map));
```
요청에서 queryParam을 주는 방법이군요!
새로운 것을 알았습니다..ㅎㅎ

102092 commented on src/test/java/com/codesquad/rare/domain/post/PostControllerTest.java:
```diff
@@ -279,4 +284,95 @@ void find_post_by_id() throws Exception {
             )));
   }
+
+  @DisplayName("포스트를 좋아요 수를 기준으로 내림차순 출력")
+  @Test
+  public void find_all_by_likes_in_descending_order() throws Exception {
```
테스트 이름은 좋아요를 기준으로 내림차순 정렬한다는 말 같은데,
내부에서는 좋아요 순으로 내림차순 정렬이 되었는지 확인하는 테스트 코드가 없는 것 같아요
이 부분을 확인하는 코드가 있었으면 좋을듯 합니다 ~

hanurii replied:
아하... 그 생각은 미처 못했네요 내림차순이 되었는지 확인하는 테스트 코드라... 이 부분은 Docs가 아닌 따로 테스트 코드가 필요한 부분이겠죠?

102092 commented:
고생하셨습니다 💯 
밤늦게 까지 하시던 것 같은데.. 너무 무리 하지 않으셨으면 합니당..ㅠㅠ
몇가지 코멘트를 남겼으니, 확인해보시고 같이 이야기 해봤으면 합니다!

hanurii committed:
[#8] Modify : Spring Rest Docs 이름의 통일성을 위해 변경

- Spring Rest Docs 이름의 통일성을 위해 변경

hanurii committed:
[#8] Modify : Integer -> int  타입으로 변경

- null 이 들어올 것을 예상했다면 랩퍼 클래스를 사용했을 것이다.
- 하지만 페이지와 사이즈는 클라이언트 단에서 숫자로 넣어줘야 하는 것이므로 int 형으로 변경

hanurii committed:
[#8] Modify : 테스트 @DisplayName 이름을 통일성있게 변경

- 가독성을 높이고자 통일성을 주었다

102092 commented on src/test/java/com/codesquad/rare/domain/post/PostControllerTest.java:
```diff
@@ -285,7 +285,7 @@ void find_post_by_id() throws Exception {
   }
 
-  @DisplayName("포스트를 좋아요 수를 기준으로 내림차순 출력")
+  @DisplayName("메인 페이지 조회 (좋아요 순으로 내름차순 정렬)")
```
내름차순....ㅋㅋㅋ
